File: leet_SLWD_1493_longest_ones.cc
class Solution {
public:
    int longestSubarray(vector<int>& n)
    {
        int tt = 1;
        int L = 0;
        int R = -1;
        while (++R < (int) n.size())
        {
            if (n[R] == 0)
                --tt;
            if (tt < 0)
            {
                if (n[L] == 0)
                    ++tt;
                ++L;
            }
        }
        return R - L - 1;
    }
};

File: leet_SLWD_1493_longest_ones.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        tt = 1
        L = 0
        for R in range(len(nums)):
            if nums[R] == 0:
                tt -= 1
            if tt < 0:
                if nums[L] == 0:
                    tt += 1
                L += 1
        return len(nums) - 1 - L

